src/main/java/teammates/common/util/FieldValidator.java

```diff
@@ -199,6 +199,11 @@
     public static final String REGEX_COURSE_ID = "[a-zA-Z0-9_.$-]+";
 
     /**
+     * Matches characters considered illegal in a course ID (inverse of above).
+     */
+    public static final String REGEX_COURSE_ID_ILLEGAL_CHARS = "[^a-zA-Z0-9_.$-]+";
```

Comment:
Is it possible to define this based on `REGEX_COURSE_ID`? Otherwise, we violate DRY principle (because 'legal chars' are defined in two places, although the 2nd place is using the inverse)